examples/compute/main.c

```diff
@@ -0,0 +1,180 @@
+#include "./../../ffi/wgpu.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+#define BINDINGS_LENGTH (1)
+#define BIND_GROUP_LAYOUTS_LENGTH (1)
+
+WGPUByteArray read_file(const char *name) {
+    FILE *file = fopen(name, "rb");
+    fseek(file, 0, SEEK_END);
+    long length = ftell(file);
+    unsigned char *bytes = malloc(length);
+    fseek(file, 0, SEEK_SET);
+    fread(bytes, 1, length, file);
+    fclose(file);
+    return (WGPUByteArray){
+        .bytes = bytes,
+        .length = length,
+    };
+}
+
+void read_buffer_map(
+    WGPUBufferMapAsyncStatus status, 
+    const uint8_t *data, 
+    uint8_t *userdata) {
+    if (status == WGPUBufferMapAsyncStatus_Success) {
+        uint32_t *times = (uint32_t *) data;
+        printf("Times: [%d, %d, %d, %d]",
+            times[0], 
+            times[1], 
+            times[2], 
+            times[3]);
+    }
+}
+
+int main(
+    int argc, 
+    char *argv[]) {
+
+    if (argc != 5) {
+        return 0;
+    }
+
+    uint32_t numbers[] = {
+        strtoul(argv[1], NULL, 0),
+        strtoul(argv[2], NULL, 0),
+        strtoul(argv[3], NULL, 0),
+        strtoul(argv[4], NULL, 0),
+    };
+
+    uint32_t size = sizeof(numbers);
+
+    uint32_t numbers_length = size / sizeof(uint32_t);
+
+    WGPUInstanceId instance = wgpu_create_instance();
+
+    WGPUAdapterId adapter = wgpu_instance_get_adapter(instance,
+        &(WGPUAdapterDescriptor){
+            .power_preference = WGPUPowerPreference_LowPower,
+        });
+
+    WGPUDeviceId device = wgpu_adapter_request_device(adapter,
+        &(WGPUDeviceDescriptor){
+            .extensions = {
```

Comment:
we should probably omit this (auto-zero-initialize) in the examples, this `extensions` thing is not very useful atm

Reply:
I am not even sure what it exactly does, copied this from the rust example. Will also update the rust example in the future with this.